### PR TITLE
support bottom level compression

### PR DESCRIPTION
**Pull request by panda-sheep**
#### What type of PR is this?
- [ ] bug
- [X] feature
- [ ] enhancement

#### What problem(s) does this PR solve?
Issue(s) number: 

Description:
support bottommost level compression

#### Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [ ] N/A

Affects:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

#### Release notes:

Please confirm whether to be reflected in release notes and how to describe:
> ex. Fixed the bug .....

### CODE REVIEW

**Review by liuyu85cn — APPROVED** [review]

File: src/kvstore/RocksEngineConfig.cpp [hunk 1]
```diff
@@ -59,7 +59,13 @@ DEFINE_bool(enable_partitioned_index_filter, false, "True for partitioned index
 DEFINE_string(rocksdb_compression,
               "snappy",
               "Compression algorithm used by RocksDB, "
-              "options: no,snappy,lz4,lz4hc,zstd,zlib,bzip2");
+              "options: no,snappy,lz4,lz4hc,zstd,zlib,bzip2,xpress");
```
Comment (liuyu85cn) [inline src/kvstore/RocksEngineConfig.cpp:62]: a space after comma?

File: src/kvstore/RocksEngineConfig.cpp [hunk 2]
```diff
@@ -59,7 +59,13 @@ DEFINE_bool(enable_partitioned_index_filter, false, "True for partitioned index
...
+
+DEFINE_string(rocksdb_bottommost_compression,
+              "disable",
+              "Specify the bottommost level compression algorithm"
+              "options: no,snappy,lz4,lz4hc,zstd,zlib,bzip2,xpress,disable");
```
Comment (liuyu85cn) [inline src/kvstore/RocksEngineConfig.cpp:67]: ditto

File: src/kvstore/RocksEngineConfig.cpp [hunk 3]
```diff
@@ -142,7 +148,9 @@ static const std::unordered_map<std::string, rocksdb::CompressionType> kCompress
     {"lz4hc", rocksdb::kLZ4HCCompression},
     {"zstd", rocksdb::kZSTD},
     {"zlib", rocksdb::kZlibCompression},
-    {"bzip2", rocksdb::kBZip2Compression}};
+    {"bzip2", rocksdb::kBZip2Compression},
+    {"xpress", rocksdb::kXpressCompression},
+    {"disable", rocksdb::kDisableCompressionOption}};
```
Comment (wenhaocs) [inline src/kvstore/RocksEngineConfig.cpp:153]: what's the difference between disable and no?

File: src/kvstore/RocksEngineConfig.cpp (hunk 1 above)
Comment (panda-sheep) [reply src/kvstore/RocksEngineConfig.cpp:62]: good catch.

File: src/kvstore/RocksEngineConfig.cpp (hunk 3 above)
Comment (panda-sheep) [reply src/kvstore/RocksEngineConfig.cpp:153]: kNoCompression means no compression
kDisableCompressionOption is used to disable some compression options.
refer to https://github.com/facebook/rocksdb/blob/fe31dc53ca04a6241a1489be826b20218dff987e/include/rocksdb/compression_type.h#L17

**Review by liuyu85cn — APPROVED** [review]

File: src/kvstore/RocksEngineConfig.cpp (hunk 3 above)
Comment (wenhaocs) [reply src/kvstore/RocksEngineConfig.cpp:153]: I see. Awesome job.

**Review by wenhaocs — APPROVED** [review]

**Pull request merged**